src/Voron/Data/Tables/TableValueCompressor.cs

```diff
@@ -214,9 +217,10 @@ private void MaybeTrainCompressionDictionary(Table table, FixedSizeTree etagsTre
                         using var compressionDictionary = new ZstdLib.CompressionDictionary(newId, dictionaryBuffer.Ptr, dictionaryBufferSpan.Length, 3);
 
                         if (ShouldReplaceDictionary(tx, compressionDictionary) == false)
-                        {
                             return;
-                        }
+
+                        if (Logger.IsInfoEnabled)
```

Comment:
how often we hit this? would like to avoid hitting this in a hot path too often

Reply:
We don't get here very often, this place is not a hot path, as we only reach this point when we need to train a new dictionary, when there are enough documents for it, and when the new dictionary is more than 10% better than the previous one.